lib/utils.coffee

```diff
@@ -310,6 +311,9 @@ requestAsync = (fetch, options, retriesRemaining) ->
 		p = p.timeout(opts.timeout)
 
 	p = p.then (response) ->
+		if opts.signal?
```

Comment:
I don't really love the `?` in case that we could just check for a falsy value (can we in this case?).
It will also make the decaffeinated code more complex.